pkg/proxy/proxy.go

```diff
@@ -130,7 +116,16 @@ func createHTTPClient(instance cfg.ServerGroup, logger log.Logger) (*http.Client
 	return client, nil
 }
 
-func ProxyHandler(config *cfg.Config, logger log.Logger) func(http.ResponseWriter, *http.Request) {
+type (
+	proxy struct {
+		config  *cfg.Config
+		logger  log.Logger
+		clients map[string]*http.Client
+	}
+	transformFn func(context.Context, http.ResponseWriter, <-chan *proxyresponse.BackendResponse, log.Logger)
+)
+
+func proxyHandler(config *cfg.Config, logger log.Logger) func(http.ResponseWriter, *http.Request) {
```

Comment:
Hi @erka did you change the function to private for any specific reason?

Reply:
As this function is only used within the package, I don’t see a strong reason to expose it publicly.